TypeCobol/Compiler/CodeElements/Entry/DataDescriptionEntry.cs

```diff
@@ -75,8 +76,28 @@ public abstract class DataDefinitionEntry: CodeElement, INamedCodeElement {
         /// </summary>
         public SymbolDefinition DataName { get; set; }
 
-        public override bool VisitCodeElement(IASTVisitor astVisitor) {
-            return base.VisitCodeElement(astVisitor) && astVisitor.Visit(this) 
+        public override TextAreaType StartingArea
+        {
+            get
+            {
+                if (ConsumedTokens == null || LevelNumber == null) //Cannot decide without ConsumedTokens
```

Comment:
Why test the `ConsumedTokens`, we only use `LevelNumber` to decide.
Also this coud be rewritten as a `switch`  or switch expression.